### PR TITLE
Handle name and container image validation in deploy form

### PR DESCRIPTION
**Pull request by bryk**
This also aligns the UX to have deploy button always active and show errors on click

### CODE REVIEW

**Comment by zreigz** [comment]
I can review this :)

**Comment by bryk** [comment]
Yes, please review :)

File: src/test/frontend/deploy/deploy_controller_test.js [hunk 1]
```diff
@@ -46,7 +43,7 @@ describe('Deploy controller', () => {
     let result = ctrl.isDeployDisabled();
 
     // then
-    expect(result).toBeTruthy();
+    expect(result).toBe(true);
```
Comment (zreigz) [inline src/test/frontend/deploy/deploy_controller_test.js]: I really like this change.  ```toBeTruthy``` was horrible for me :) I will also start using this ```toBe(true)```

**Comment by zreigz** [comment]
I've checked this. Everything is displayed correctly. LGTM

**Comment by bryk** [comment]
Thanks!

**Pull request merged**